src/Npgsql/NpgsqlSlimDataSourceBuilder.cs

```diff
@@ -375,6 +378,50 @@ static string GetPgName(Type clrType, INpgsqlNameTranslator nameTranslator)
 
     #endregion Type mapping
 
+    #region Optional opt-ins
+
+    /// <summary>
+    /// Sets up mappings for the PostgreSQL <c>range</c> and <c>multirange</c> types.
+    /// </summary>
+    public NpgsqlSlimDataSourceBuilder EnableRanges()
+    {
+        AddTypeResolverFactory(new RangeTypeHandlerResolverFactory());
+        return this;
+    }
+
+    /// <summary>
+    /// Sets up System.Text.Json mappings for the PostgreSQL <c>json</c> and <c>jsonb</c> types.
+    /// </summary>
+    /// <param name="serializerOptions">Options to customize JSON serialization and deserialization.</param>
+    /// <param name="jsonbClrTypes">
+    /// A list of CLR types to map to PostgreSQL <c>jsonb</c> (no need to specify <see cref="NpgsqlDbType.Jsonb" />).
+    /// </param>
+    /// <param name="jsonClrTypes">
+    /// A list of CLR types to map to PostgreSQL <c>json</c> (no need to specify <see cref="NpgsqlDbType.Json" />).
+    /// </param>
+    public NpgsqlSlimDataSourceBuilder UseSystemTextJson(
+        JsonSerializerOptions? serializerOptions = null,
+        Type[]? jsonbClrTypes = null,
+        Type[]? jsonClrTypes = null)
+    {
+        AddTypeResolverFactory(new JsonTypeHandlerResolverFactory(jsonbClrTypes, jsonClrTypes, serializerOptions));
+        return this;
+    }
+
+    /// <summary>
+    /// Enables to possibility to use TLS/SSl encryption for connections to PostgreSQL. This does not guarantee that encryption will
+    /// actually be used; see <see href="https://www.npgsql.org/doc/security.html"/> for more details.
+    /// </summary>
+    public NpgsqlSlimDataSourceBuilder EnableEncryption()
```

Comment:
We could just call it Add* instead of enable

Reply:
I looked over it the naming again... Most of the other APIs on the builder are called Use*; but they actually mean that the feature will be used (e.g. UseClientCertificate, UseLoggerFactory). There's EnableParameterLogging, which is about turning a feature on or off (UseParameterLogging makes no sense), and also aligns with EF (in case that matters.

I *think* EnableEncryption and EnableRanges is consistent - it's just turning a feature on... SystemTextJson is annoying, since it's both turning on the possibility to use JSON (if using the slim builder) and also configuring it with serializer options etc. (which is why it's also exposed on the non-slim builder).

Long story short... Add also seems fine; I admit I slightly prefer Enable since it's very clearly about turning something on/off; but it's also consistent with the existing EnableParameterLogging, which is also a point in its favor...

I'll go ahead and merge this for now, but am perfectly ok rediscussing the naming and changing.